Assets/HoloToolkit-Preview/QRTracker/Scripts/QRCode.cs

```diff
@@ -0,0 +1,98 @@
+    {
+        public QRCodesTrackerPlugin.QRCode qrCode;
+        private GameObject qrCodeCube;
+
+        public float PhysicalSize;
+        public string CodeText { get; set; }
+
+        private TextMesh QRID;
+        private TextMesh QRText;
+        private TextMesh QRVersion;
+        private TextMesh QRTimeStamp;
+        private TextMesh QRSize;
+        private GameObject QRInfo;
+
+        private long lastTimeStamp = 0;
+
+        // Use this for initialization
+        void Start()
+        {
+            PhysicalSize = 0.1f;
+            CodeText = "Dummy";
+            if (qrCode == null)
+            {
+                throw new System.Exception("QR Code Empty");
+            }
+
+            PhysicalSize = qrCode.PhysicalSizeMeters;
+            CodeText = qrCode.Code;
+
+            qrCodeCube = gameObject.transform.Find("Cube").gameObject;
+            QRInfo = gameObject.transform.Find("QRInfo").gameObject;
+            QRID = QRInfo.transform.Find("QRID").gameObject.GetComponent<TextMesh>();
+            QRText = QRInfo.transform.Find("QRText").gameObject.GetComponent<TextMesh>();
+            QRVersion = QRInfo.transform.Find("QRVersion").gameObject.GetComponent<TextMesh>();
+            QRTimeStamp = QRInfo.transform.Find("QRTimeStamp").gameObject.GetComponent<TextMesh>();
+            QRSize = QRInfo.transform.Find("QRSize").gameObject.GetComponent<TextMesh>();
+
+            QRID.text = qrCode.Id.ToString();
+            QRText.text = CodeText;
+            QRVersion.text = "Ver: " + qrCode.Version;
+            QRSize.text = "Size:" + qrCode.PhysicalSizeMeters.ToString("F04") + "m";
+            QRTimeStamp.text = "Time:" + qrCode.LastDetectedQPCTicks;
+            Debug.Log("Id= " + qrCode.Id + " PhysicalSize = " + PhysicalSize + " TimeStamp = " + qrCode.LastDetectedQPCTicks + " QRVersion = " + qrCode.Version + " QRData = " + CodeText);
+        }
+
+        void UpdatePropertiesDisplay()
+        {
+            // Update properties that change
+            if (qrCode != null && lastTimeStamp != qrCode.LastDetectedQPCTicks)
+            {
+                QRSize.text = "Size:" + qrCode.PhysicalSizeMeters.ToString("F04") + "m";
+#if true
+                {
+                    long EndingTime = System.Diagnostics.Stopwatch.GetTimestamp();
+                    long ElapsedTime = EndingTime - (long)qrCode.LastDetectedQPCTicks;
+
+                    double ElapsedSecs = ElapsedTime * (1.0f / System.Diagnostics.Stopwatch.Frequency);
+
+                    QRTimeStamp.text = "Time:" + System.DateTime.Now.AddSeconds(-ElapsedSecs).ToString("MM/dd/yyyy HH:mm:ss.fff");
+
+                }
+#else
+                {
+                    QRTimeStamp.text = "Time:" + qrCode.LastDetectedQPCTicks;
+                }
+#endif
+
+                PhysicalSize = qrCode.PhysicalSizeMeters;
+                Debug.Log("Id= " + qrCode.Id + " PhysicalSize = " + PhysicalSize + " TimeStamp = " + qrCode.LastDetectedQPCTicks);
```

Comment:
nit: I _think_ string interpolation does less allocations than string +

Reply:
Old HTK does not use c# 4 language features. I think the .NET support is only 3.5

Reply:
Ah! I missed that this was for HTK only